test/jdk/java/awt/Window/OwnedWindowShowTest/OwnedWindowShowTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 1998, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4177156
+ * @key headful
+ * @summary Tests that multiple level of window ownership doesn't cause
+ * NullPointerException when showing a Window
+ * @run main OwnedWindowShowTest
+ */
+
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Window;
+
+public class OwnedWindowShowTest {
+    public static void main(String args[]) throws Exception {
+        EventQueue.invokeAndWait(() -> runTest());
+    }
+
+    static void runTest() {
+        Frame parent = new Frame("OwnedWindowShowTest");
+        try {
+            Window owner = new Window(parent);
+            Window window = new Window(owner);
+            window.setVisible(true);
+        } finally {
+            if (parent != null) {
+                parent.dispose();
+            }
```

Comment:
The condition `parent != null` is always `true` when `if`-statement is reached.
```suggestion
            parent.dispose();
```

You may also want to dispose of `window` and `owner` too.

Reply:
> If parent is disposed, i expect child components also to be disposed. I don't see any UI components after test is finished. Do we need to explicitly dispose all child components?

You're right: disposing of a window also disposes of all the owned windows.

https://github.com/openjdk/jdk/blob/db61458da840123925cb3ba079cfaf8277880320/src/java.desktop/share/classes/java/awt/Window.java#L1199-L1209